### PR TITLE
Use custom template on ScanCode analyzer

### LINKED ISSUE #464: Use a custom output format on Scancode
Currently, scancode analyzer uses the default json output from scancode.

This requires further processing/parsing on the go side to bring the data in the format required by the protobuf spec. The main issue with that is that it's done by iterating through the parsed file list for each fileNode received from the master service.

Using the path as a key in the custom structure we can reduce the computational complexity by using a map-key access with the fileNode's path.

Also it allows us to create a go structure that conforms both to
the custom output and the protobuf specs.

REF:

Scancode custom output: https://scancode-toolkit.readthedocs.io/en/latest/tutorials/how_to_format_scan_output.html#custom-output-format
Jinja2 docs: https://jinja.palletsprojects.com/en/2.11.x/U

### PR DESCRIPTION
**Pull request by GiasemiSh**
closes #464 

* We need to update the demos. The workdir in the scancode yaml configuration needs to be the dir to the source code of the project.
* Check also the doc demos

### CODE REVIEW

File: modules/analyzers/scancode-analyzer/main.go [hunk 1]
```diff
@@ -147,122 +145,64 @@ func scancodeExitHandler(err error, output []byte) {
...
 		resultfilepath = tmpfile.Name()
 		if err != nil {
-			return nil, err
+			return err
 		}
 		defer os.Remove(tmpfile.Name())
 	}
 
-	cmdlineargs := []string{"--full-root", "-l", "-c", "--json", resultfilepath}
+	cmdlineargs := []string{"--full-root", "-l", "-c", "--custom-output", resultfilepath, "--custom-template", "qmstr.j2"}
```
Comment (pombredanne) [inline modules/analyzers/scancode-analyzer/main.go]: Note that this `--custom-output` option is a slightly less traveled area, so there may be some dragons here and there... and there are some things that you cannot get from that such as the detected license texts (IMHO you should use the `--license-text` ) 
Also I see that you may be missing the --package option to collect package manifests and their licenses.

Comment (GiasemiSh) [reply modules/analyzers/scancode-analyzer/main.go]: Thanks for the input @pombredanne! So your opinion is to skip the `--custom-output` flag? You believe that we are going to miss some license information when using the flag?

**Comment by GiasemiSh** [comment]
Depends on #478. Waiting feedback.

**Comment by GiasemiSh** [comment]
Since `--custom-output` option is a less traveled area and we are missing some information from the default function, we will not use the `--custom-output` option for now.

**Pull request closed without merging**